addons/l10n_in/models/account_move_line.py

```diff
@@ -10,4 +10,5 @@ class AccountMoveLine(models.Model):
     def _compute_l10n_in_hsn_code(self):
         for line in self:
             if line.move_id.country_code == 'IN' and line.parent_state == 'draft':
-                line.l10n_in_hsn_code = line.product_id.l10n_in_hsn_code
+                if line.product_id and line.product_id.l10n_in_hsn_code:
+                    line.l10n_in_hsn_code = line.product_id.l10n_in_hsn_code[:int(line.company_id.l10n_in_hsn_code_digit)] if line.company_id.l10n_in_hsn_code_digit else line.product_id.l10n_in_hsn_code
```

Comment:
For @tsb-odoo and me, this change of behaviour is weird: it means that less digits is put when you actually have more digits and you can provide more clarity, so we think it is confusing for the user.  The case where you explicitly want to put less digits than what you have on the product seems to be a corner case and as we have the field on the account move line, the user can still adapt himself explicitly.  I think it would be better to not have this change in _compute_l10n_in_hsn_code, also because suddenly it will work differently in existing databases.  @ajjo-odoo  @zepa-odoo